src/Auth0.OidcClient.Android/ChromeCustomTabsBrowser.cs

```diff
@@ -29,6 +29,13 @@ protected override void OpenBrowser(Android.Net.Uri uri, Context context = null)
             {
                 if (IsNewTask)
                     customTabsIntent.Intent.AddFlags(ActivityFlags.NewTask);
+
+                if (AutoCloseBrowser)
+                {
+                    customTabsIntent.Intent.AddFlags(ActivityFlags.NoHistory);
```

Comment:
What's the purpose of using `NoHistory` here ? As I understand correctly, `ClearTop` should be the one solving the issue, or is this incorrect?

Reply:
The reason I included the "NoHistory" flag was based on the description found here: https://developer.android.com/guide/topics/manifest/activity-element#nohist
I did not want the Auth0 window to be accessible with the "Back" command in Android. Even if it Auth0 activity had been closed, a new would be created, letting the user come back to the login screen.

The FLAG_ACTIVITY_CLEAR_TOP is used for clearing any other activities on the navigation stack and return navigation to the initial (old) activity and then let it receive the intent.

Reply:
I am curious to understand what the intention is for adding `NoHistory` here. How does it relate to the problem mentioned about the browser not being closed.

What is the actual issue/risk you see with the login page being part of the history?

Reply:
By leaving in the history the login page, the users could end up going "back" to the login page and login again, without a proper logout first.
There are some behavior questions that need to be answered for this, like

- What happens with the app if the user does another login?
- What happens with the app if the user login with a different account?
- How do we manage the transition "Back" to the login page? Do we prompt the user? Do we auto-logout? What if the "back" was an accident?

The only way to avoid this easily, is to only be able to logout and then login again - and to make sure one user does not have access to the previous users information, is to clear the navigation history.

Reply:
All our browser-based SDKs (https://github.com/auth0/auth0-spa-js, https://github.com/auth0/auth0-angular, https://github.com/auth0/auth0-react) as well as our mobile Java SDK (https://github.com/auth0/Auth0.Android/blob/main/auth0/src/main/java/com/auth0/android/provider/AuthenticationActivity.kt#L101) end up having the login page end up in history, which is why I am not sure we should add it here. 

The problems you describe span all of these SDKs and haven't shown to be an actual issue so far.

The intention here is solving an issue that seems to be unrelated to removing the login page from the browser history, if you believe this is an issue I would recommend opening an issue so we can take this internally to discuss with all of the maintainers of the mentioned SDKs that are impacted by the login page ending up in the browser's history.

Reply:
It might be an edge case that no-one else have given much thought. I do not know if this is an issue for Xamarin.droid only, but I do not think it is. I strongly believe there is a genuine security risk for being able to go back to the login screen and basically be able to go "forward" again, not knowing who is logged in.
If the decision is to leave out that part of it, I would strongly recommend a solution where the flag can be added or removed by choice.
I will create a separate issue for this.

Reply:
I am happy to raise your concern internally with our security team.

For what I know, once u are logged in with Auth0 you have a session with our Auth0 server. Hitting back visits the login page, identifies u have a session and instantly redirects you. 

Before being able to actually raise it internally, I would like to gather some more information about the security risk you are seeing. Would you happen to want to reflect your concern in the created issue (https://github.com/auth0/auth0-oidc-client-net/issues/204) rather than mentioning you where asked to create an issue ?  This will help us driving the conversation with the security team.

Thanks 🙏

Reply:
You are welcome, I appreciate you appreciating it!

This library is great and I am happy to be to continue to use it - and if I can help making it better in any way, even if it is just for the sake of the argument and making sure the concern has been addressed and given some thought. I could easily be wrong in my assessment of the risk, but we can't know until we have talked about it.

I will remove the "NoHistory" flag from the PR and then leave it up to issue #204 to have the other part addressed, if needed.

**Did you want me to keep the backward compatibility (extra code added to keep the default behavior as it is now) or do you want me to just add the extra flag (ActivityFlags.ClearTop) to close the browser (and then change the default behavior of the library)?**

Reply:
> Did you want me to keep the backward compatibility (extra code added to keep the default behavior as it is now) or do you want me to just add the extra flag (ActivityFlags.ClearTop) to close the browser (and then change the default behavior of the library)?

Will get back to this asap, sorry about that delay. Feel free to keep the code as is for now.